contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -4418,3 +4418,45 @@ BEGIN
 END;
 $body$
 LANGUAGE plpgsql IMMUTABLE;
+
+CREATE OR REPLACE FUNCTION SYS.TYPE_NAME(IN type_id INT)
+RETURNS SYSNAME AS $$
+    BEGIN
+        RETURN (SELECT ty.name FROM sys.types ty WHERE ty.user_type_id = type_id);
+    END;
+$$
+LANGUAGE plpgsql IMMUTABLE
+STRICT;
+
+CREATE OR REPLACE FUNCTION SYS.TYPE_ID(IN typename SYS.NVARCHAR)
+RETURNS INT AS $$
+    DECLARE
+        schemaid int;
+        schema_name VARCHAR;
+        type_name VARCHAR;
+    BEGIN
+        IF typename LIKE '%.%' THEN
+            schema_name := TRIM(LOWER(split_part(typename COLLATE "C", '.', 1)));
+            type_name :=  TRIM(LOWER((split_part(typename COLLATE "C",'.', 2))));
+        ELSE
+            schema_name := 'dbo';
+            type_name := TRIM(LOWER(typename COLLATE "C"));
+        END IF;
+
+        -- Check if schema_name exists
+        IF NOT EXISTS (SELECT sc.name FROM sys.schemas sc WHERE sc.name = schema_name OR schema_name = 'sys' OR schema_name = 'pg_catalog')
```

Comment:
can we use our sys extended catalog like babelfish_namespace_ext? it would faster than this reference to view